sdks/go/pkg/beam/core/typex/special.go

```diff
@@ -64,6 +65,19 @@ type Window interface {
 	Equals(o Window) bool
 }
 
+const (
+	PaneEarly   int = 0 // EARLY
+	PaneOnTime  int = 1 // ON_TIME
+	PaneLate    int = 2 // LATE
+	PaneUnknown int = 3 // UNKNOWN
+)
+
+type PaneInfo struct {
+	Timing                     int
```

Comment:
As much as it's probably extra, we don't want this to be an int in type, as then users might try to manipulate it as an int.
Instead add a new type `type PaneTiming byte`  (since we really won't need more than that presently, or likely, ever) then the above enum declaration can be

```
const (
	PaneEarly   PaneTiming = 0 
	PaneOnTime  PaneTiming = 1 
	PaneLate    PaneTiming = 2 
	PaneUnknown PaneTiming = 3 
)
```
(Since we have specific definitions, we can't lean on an iota, even though it comes out the same. We want them to match their defined meaning in the coder definition).

The net result is that in order to use it in math/etc, a user would need to explicitly type cast to a numeric value. This avoids being able to misuse the enum type as a value rather than symbolically which is all it needs to have. 

This is necessary since the PaneInfo will eventually exposed to user DoFns, and it needs to be hard to misuse, which we're doing with the type system.